h2o-core/src/main/java/water/H2O.java

```diff
@@ -1324,6 +1325,9 @@ public static String DEFAULT_FLOW_DIR() {
    * It is updated also when a new client appears. */
   private static HashSet<H2ONode> STATIC_H2OS = null;
 
+  /* List of all clients that ever connected to this cloud */
+  private static Map<H2ONode.H2Okey, H2ONode> CLIENTS_MAP = new ConcurrentHashMap<>();
```

Comment:
ah, right. I didn't know we have our own implementation. I will keep it in my mind for the next time! Thanks!